### PR TITLE
Improve oci-sysext to handle optional image-source

### PR DESCRIPTION
**Pull request by krishjainx**
Improve createRootfs to handle optional image-source

Improve the createRootfs function to correctly manage layer skipping when the image-source is the same as the image or not provided. This update ensures proper extraction of OCI image layers based on differential or full image usage.

**Note: Updated deps as well. Ignore updates in `vendor` folder**

### CODE REVIEW

**Comment by krishjainx** [comment]
cc/ @89luca89 also updated dependencies.

**Comment by krishjainx** [comment]
Currently the best we're able to do with oci-sysext is this

```
(base) krishjain@notubuntu:~/oci-sysext$ git log -1
commit c4b7457d776b74fd89f620a68c8d4f0a6b436f59 (HEAD -> fix-up, origin/fix-up)
Author: Krish Jain <kjain7@u.rochester.edu>
Date:   Fri Jun 21 01:16:37 2024 -0400

    Updated README
(base) krishjain@notubuntu:~/oci-sysext$  sudo rm -rf /root/.local/share/oci-sysext/ && go build . && sudo  ./oci-sysext create --image cgr.dev/chainguard/wolfi-base --name zero-cve-userspace --fs btrfs 
sysext_utils.go:190 [info] cleaning up rootfs dir...
sysext_utils.go:196 [info] ensuring image cgr.dev/chainguard/wolfi-base ...
pulling image manifest: cgr.dev/chainguard/wolfi-base:latest
image_utils.go:280 [info] pulling layer 7bb5c1d3e7b44ef4f9307e2e8e88d0a65e5333c612c2bc980dded942a06776d3.tar.gz
Copying blob sha256:7bb5c1d3e7b44ef4f9307e2e8e88d0a65e5333c612c2bc980dded942a06776d3 100% |██████████████████████████████| (27 MB/s)         
image_utils.go:337 [info] saving layer sha256:7bb5c1d3e7b44ef4f9307e2e8e88d0a65e5333c612c2bc980dded942a06776d3 done
saving manifest for cgr.dev/chainguard/wolfi-base:latest
saving config for cgr.dev/chainguard/wolfi-base:latest
saving metadata for cgr.dev/chainguard/wolfi-base:latest
done
sysext_utils.go:90 [info] preparing rootfs for new sysext zero-cve-userspace
sysext_utils.go:98 [info] creating /root/.local/share/oci-sysext/sysexts-rootfs/8dd682da869e52af509cf237f35540ca
sysext_utils.go:105 [info] looking up image cgr.dev/chainguard/wolfi-base
sysext_utils.go:107 [info] reading cgr.dev/chainguard/wolfi-base's manifest
sysext_utils.go:119 [info] extracting image's layers, skipping 0 layers...
sysext_utils.go:131 [info] extracting layer 7bb5c1d3e7b44ef4f9307e2e8e88d0a65e5333c612c2bc980dded942a06776d3.tar.gz in /root/.local/share/oci-sysext/sysexts-rootfs/8dd682da869e52af509cf237f35540ca
sysext_utils.go:146 [info] removing unneeded dir: bin
sysext_utils.go:146 [info] removing unneeded dir: dev
sysext_utils.go:146 [info] removing unneeded dir: etc
sysext_utils.go:146 [info] removing unneeded dir: home
sysext_utils.go:146 [info] removing unneeded dir: lib
sysext_utils.go:146 [info] removing unneeded dir: lib64
sysext_utils.go:146 [info] removing unneeded dir: proc
sysext_utils.go:146 [info] removing unneeded dir: root
sysext_utils.go:146 [info] removing unneeded dir: run
sysext_utils.go:146 [info] removing unneeded dir: sbin
sysext_utils.go:146 [info] removing unneeded dir: tmp
sysext_utils.go:146 [info] removing unneeded dir: var
sysext_utils.go:165 [info] rootfs creation done
sysext_utils.go:218 [info] creating raw file
(base) krishjain@notubuntu:~/oci-sysext$ sudo su
root@notubuntu:/home/krishjain/oci-sysext# /root/.local/share/oci-sysext/sysexts-rootfs/8dd682da869e52af509cf237f35540ca/bin/busybox 
BusyBox v1.36.1 (2024-05-23 13:45:30 UTC) multi-call binary.
BusyBox is copyrighted by many authors between 1998-2015.
Licensed under GPLv2. See source distribution for detailed
copyright notices.

Usage: busybox [function [arguments]...]
   or: busybox --list[-full]
   or: busybox --install [-s] [DIR]
   or: function [arguments]...

	BusyBox is a multi-call binary that combines many common Unix
	utilities into a single executable.  Most people will create a
	link to busybox for each function they wish to use and BusyBox
	will act like whatever it was invoked as.

Currently defined functions:
	[, [[, add-shell, addgroup, adduser, adjtimex, arch, arping, ash, awk, base64, basename, bbconfig, bc, beep, bunzip2, bzcat, bzip2, cal, cat, chattr, chgrp, chmod, chown, chpasswd, chroot, chrt,
	cksum, clear, cmp, comm, cp, cpio, cryptpw, cut, date, dc, dd, delgroup, deluser, df, diff, dirname, dmesg, dnsdomainname, dos2unix, du, echo, ed, egrep, env, expand, expr, factor, fallocate,
	false, fgrep, find, findfs, flock, fold, free, fsync, fuser, getopt, getty, grep, groups, gunzip, gzip, hd, head, hexdump, hostid, hostname, id, inotifyd, install, ionice, iostat, ipcrm, ipcs,
	kill, killall, killall5, less, link, linux32, linux64, ln, logger, login, ls, lsattr, lsof, lzcat, lzma, lzop, lzopcat, md5sum, microcom, mkdir, mkfifo, mknod, mkpasswd, mktemp, more,
	mountpoint, mpstat, mv, netstat, nice, nl, nmeter, nohup, nologin, nproc, nsenter, od, passwd, paste, pgrep, pidof, ping, ping6, pipe_progress, pivot_root, pkill, pmap, printenv, printf, ps,
	pstree, pwd, pwdx, rdev, readahead, readlink, realpath, remove-shell, renice, reset, resize, rev, rm, rmdir, run-parts, sed, seq, setpriv, setserial, setsid, sh, sha1sum, sha256sum, sha3sum,
	sha512sum, shred, shuf, sleep, sort, split, stat, strings, stty, su, sum, sync, sysctl, tac, tail, tar, tee, test, time, timeout, top, touch, tr, traceroute, traceroute6, tree, true, truncate,
	tsort, tty, ttysize, tunctl, uname, unexpand, uniq, unix2dos, unlink, unlzma, unlzop, unxz, unzip, uptime, usleep, uudecode, uuencode, vconfig, vi, vlock, watch, wc, which, who, whoami, xargs,
	xxd, xzcat, yes, zcat
root@notubuntu:/home/krishjain/oci-sysext# /root/.local/share/oci-sysext/sysexts-rootfs/8dd682da869e52af509cf237f35540ca/bin/busybox sh
/home/krishjain/oci-sysext # 
root@notubuntu:/home/krishjain/oci-sysext# /root/.local/share/oci-sysext/sysexts-rootfs/8dd682da869e52af509cf237f35540ca/usr/bin/awk 
bash: /root/.local/share/oci-sysext/sysexts-rootfs/8dd682da869e52af509cf237f35540ca/usr/bin/awk: cannot execute binary file: Exec format error
root@notubuntu:/home/krishjain/oci-sysext# ldd /root/.local/share/oci-sysext/sysexts-rootfs/8dd682da869e52af509cf237f35540ca/bin/busybox
	linux-vdso.so.1 (0x00007ffd1935f000)
	libcrypt.so.1 => /lib64/libcrypt.so.1 (0x00007fc253640000)
	libm.so.6 => /lib64/libm.so.6 (0x00007fc25355d000)
	libc.so.6 => /lib64/libc.so.6 (0x00007fc253370000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fc25372d000)
root@notubuntu:/home/krishjain/oci-sysext# ldd /root/.local/share/oci-sysext/sysexts-rootfs/8dd682da869e52af509cf237f35540ca/usr/bin/awk
	linux-vdso.so.1 (0x00007ffe3e32a000)
	libc.musl-x86_64.so.1 => not found

```

**Comment by krishjainx** [comment]
Debian container fyi:

```
root@notubuntu:/home/krishjain/oci-sysext# ldd /root/.local/share/oci-sysext/sysexts-rootfs/6e9552c9bd8e61c8f277c21220160234/usr/bin/apt
    linux-vdso.so.1 (0x00007ffca5f1d000)
    libapt-private.so.0.0 => /lib64/libapt-private.so.0.0 (0x00007fd534ec0000)
    libapt-pkg.so.6.0 => /lib64/libapt-pkg.so.6.0 (0x00007fd534c00000)
    libstdc++.so.6 => /lib64/libstdc++.so.6 (0x00007fd534800000)
    libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007fd534e92000)
    libc.so.6 => /lib64/libc.so.6 (0x00007fd534613000)
    libz.so.1 => /lib64/libz.so.1 (0x00007fd534e6f000)
    libbz2.so.1 => /lib64/libbz2.so.1 (0x00007fd534e5b000)
    liblzma.so.5 => /lib64/liblzma.so.5 (0x00007fd534e28000)
    liblz4.so.1 => /lib64/liblz4.so.1 (0x00007fd534e07000)
 ......
root@notubuntu:/home/krishjain/oci-sysext# /root/.local/share/oci-sysext/sysexts-rootfs/6e9552c9bd8e61c8f277c21220160234/usr/bin/apt
apt 2.9.2 (amd64)
Usage: apt [options] command

apt is a commandline package manager and provides commands for
searching and managing as well as querying information about packages.
It provides the same functionality as the specialized APT tools,
like apt-get and apt-cache, but enables options more suitable for
interactive use by default.

Most used commands:
  list - list packages based on package names
  search - search in package descriptions
  show - show package details
  install - install packages
......

```


but with bash


```

root@notubuntu:/home/krishjain/oci-sysext#  /root/.local/share/oci-sysext/sysexts-rootfs/6e9552c9bd8e61c8f277c21220160234/usr/bin/bash
/root/.local/share/oci-sysext/sysexts-rootfs/6e9552c9bd8e61c8f277c21220160234/usr/bin/bash: /lib64/libtinfo.so.6: no version information available (required by /root/.local/share/oci-sysext/sysexts-rootfs/6e9552c9bd8e61c8f277c21220160234/usr/bin/bash)

```


This is without patching. Just with this PR/original code at time of writing. Symbol conflict. Apt doesn't use libtinfo

**Comment by krishjainx** [comment]
Just the regular code at this time/this PR and wolfi container (busybox works) everything in /usr/bin doesn't since it links to /bin/busybox on host :/

```
root@notubuntu:/home/krishjain/oci-sysext# ls -al /root/.local/share/oci-sysext/sysexts-rootfs/8dd682da869e52af509cf237f35540ca
total 4
drwxr-xr-x. 1 root root  96 Jun 22 22:05 .
drwxr-xr-x. 1 root root  64 Jun 22 22:05 ..
drwxr-xr-x. 1 root root 746 Dec 31  1969 bin
drwxr-xr-x. 1 root root   0 Dec 31  1969 dev
drwxr-xr-x. 1 root root 328 Dec 31  1969 etc
drwxr-xr-x. 1 root root  14 Dec 31  1969 home
drwxr-xr-x. 1 root root 594 Dec 31  1969 lib
lrwxrwxrwx. 1 root root   3 Dec 31  1969 lib64 -> lib
drwxr-xr-x. 1 root root   0 Dec 31  1969 opt
dr-xr-xr-x. 1 root root   0 Dec 31  1969 proc
drwx------. 1 root root   0 Dec 31  1969 root
drwxr-xr-x. 1 root root   0 Dec 31  1969 run
drwxr-xr-x. 1 root root 134 Dec 31  1969 sbin
drwxrwxrwt. 1 root root   0 Dec 31  1969 tmp
drwxr-xr-x. 1 root root  56 Dec 31  1969 usr
drwxr-xr-x. 1 root root  56 Dec 31  1969 var
root@notubuntu:/home/krishjain/oci-sysext# ls -al /root/.local/share/oci-sysext/sysexts-rootfs/8dd682da869e52af509cf237f35540ca/usr/bin/
total 476
drwxr-xr-x. 1 root root 1222 Dec 31  1969  .
drwxr-xr-x. 1 root root   56 Dec 31  1969  ..
lrwxrwxrwx. 1 root root   12 Dec 31  1969 '[' -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969 '[[' -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  awk -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  basename -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  hostid -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  id -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  install -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  ipcrm -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  ipcs -> /bin/busybox
.....
root@notubuntu:/home/krishjain/oci-sysext#  /root/.local/share/oci-sysext/sysexts-rootfs/8dd682da869e52af509cf237f35540ca/bin/busybox 
BusyBox v1.36.1 (2024-05-23 13:45:30 UTC) multi-call binary.
BusyBox is copyrighted by many authors between 1998-2015.
Licensed under GPLv2. See source distribution for detailed
copyright notices.

Usage: busybox [function [arguments]...]
   or: busybox --list[-full]
   or: busybox --install [-s] [DIR]
   or: function [arguments]...

	BusyBox is a multi-call binary that combines many common Unix
	utilities into a single executable.  Most people will create a
	link to busybox for each function they wish to use and BusyBox
	will act like whatever it was invoked as.

Currently defined functions:
	[, [[, add-shell, addgroup, adduser, adjtimex, arch, arping, ash, awk, base64, basename, bbconfig, bc, beep, bunzip2, bzcat, bzip2, cal, cat, chattr, chgrp,
	chmod, chown, chpasswd, chroot, chrt, cksum, clear, cmp, comm, cp, cpio, cryptpw, cut, date, dc, dd, delgroup, deluser, df, diff, dirname, dmesg,
	dnsdomainname, dos2unix, du, echo, ed, egrep, env, expand, expr, factor, fallocate, false, fgrep, find, findfs, flock, fold, free, fsync, fuser, getopt,
	getty, grep, groups, gunzip, gzip, hd, head, hexdump, hostid, hostname, id, inotifyd, install, ionice, iostat, ipcrm, ipcs, kill, killall, killall5, less,
	link, linux32, linux64, ln, logger, login, ls, lsattr, lsof, lzcat, lzma, lzop, lzopcat, md5sum, microcom, mkdir, mkfifo, mknod, mkpasswd, mktemp, more,
	mountpoint, mpstat, mv, netstat, nice, nl, nmeter, nohup, nologin, nproc, nsenter, od, passwd, paste, pgrep, pidof, ping, ping6, pipe_progress, pivot_root,
	pkill, pmap, printenv, printf, ps, pstree, pwd, pwdx, rdev, readahead, readlink, realpath, remove-shell, renice, reset, resize, rev, rm, rmdir, run-parts,
	sed, seq, setpriv, setserial, setsid, sh, sha1sum, sha256sum, sha3sum, sha512sum, shred, shuf, sleep, sort, split, stat, strings, stty, su, sum, sync,
	sysctl, tac, tail, tar, tee, test, time, timeout, top, touch, tr, traceroute, traceroute6, tree, true, truncate, tsort, tty, ttysize, tunctl, uname,
	unexpand, uniq, unix2dos, unlink, unlzma, unlzop, unxz, unzip, uptime, usleep, uudecode, uuencode, vconfig, vi, vlock, watch, wc, which, who, whoami, xargs,
	xxd, xzcat, yes, zcat
root@notubuntu:/home/krishjain/oci-sysext#  /root/.local/share/oci-sysext/sysexts-rootfs/8dd682da869e52af509cf237f35540ca/bin/busybox sh
/home/krishjain/oci-sysext # 
/home/krishjain/oci-sysext # 
/home/krishjain/oci-sysext # 
/home/krishjain/oci-sysext # 
root@notubuntu:/home/krishjain/oci-sysext# /root/.local/share/oci-sysext/sysexts-rootfs/8dd682da869e52af509cf237f35540ca/usr/bin/awk # stuff that is symlinked to wrong path to busybox
bash: /root/.local/share/oci-sysext/sysexts-rootfs/8dd682da869e52af509cf237f35540ca/usr/bin/awk: No such file or directory
root@notubuntu:/home/krishjain/oci-sysext# ls /root/.local/share/oci-sysext/sysexts-rootfs/8dd682da869e52af509cf237f35540ca/usr/bin/awk
/root/.local/share/oci-sysext/sysexts-rootfs/8dd682da869e52af509cf237f35540ca/usr/bin/awk
root@notubuntu:/home/krishjain/oci-sysext# ls -al /root/.local/share/oci-sysext/sysexts-rootfs/8dd682da869e52af509cf237f35540ca/usr/bin/awk
lrwxrwxrwx. 1 root root 12 Dec 31  1969 /root/.local/share/oci-sysext/sysexts-rootfs/8dd682da869e52af509cf237f35540ca/usr/bin/awk -> /bin/busybox
root@notubuntu:/home/krishjain/oci-sysext# ldd /root/.local/share/oci-sysext/sysexts-rootfs/8dd682da869e52af509cf237f35540ca/usr/bin/awk
ldd: /root/.local/share/oci-sysext/sysexts-rootfs/8dd682da869e52af509cf237f35540ca/usr/bin/awk: No such file or directory
```

**Comment by krishjainx** [comment]
Now with this PR testing Alpine. Same symlink situation as earlier (to /bin/busybox on host) but now running busybox causes ``Segmentation fault`` 🍿 

```
root@notubuntu:/home/krishjain/oci-sysext# ls /root/.local/share/oci-sysext/sysexts-rootfs/7c9fb847d117531433435b68b61f91f6/usr/bin/ -al
total 740
drwxr-xr-x. 1 root root  1516 Jun 18 10:16  .
drwxr-xr-x. 1 root root    40 Jun 18 10:16  ..
lrwxrwxrwx. 1 root root    12 Jun 18 10:16 '[' -> /bin/busybox
lrwxrwxrwx. 1 root root    12 Jun 18 10:16 '[[' -> /bin/busybox
lrwxrwxrwx. 1 root root    12 Jun 18 10:16  awk -> /bin/busybox
....................

root@notubuntu:/home/krishjain/oci-sysext# /root/.local/share/oci-sysext/sysexts-rootfs/7c9fb847d117531433435b68b61f91f6/bin/busybox 
Segmentation fault (core dumped)
root@notubuntu:/home/krishjain/oci-sysext# /root/.local/share/oci-sysext/sysexts-rootfs/7c9fb847d117531433435b68b61f91f6/bin/busybox 
Segmentation fault (core dumped)
root@notubuntu:/home/krishjain/oci-sysext# ldd /root/.local/share/oci-sysext/sysexts-rootfs/7c9fb847d117531433435b68b61f91f6/bin/busybox
	linux-vdso.so.1 (0x00007fff971f1000)
	libc.musl-x86_64.so.1 => not found
```

**Comment by krishjainx** [comment]
So for anyone reading this: FYI, we (Chainguard) do our Wolfi builds using glibc, not musl (unlike Alpine). This is why busybox was working on Wolfi (not using musl) but not on Alpine. In both cases, it was trying to use the host library (in this case working for Wolfi since I don't have a musl host).

I completely forgot about wolfi not using musl (currently).

```
root@notubuntu:/home/krishjain/oci-sysext# ldd /root/.local/share/oci-sysext/sysexts-rootfs/8dd682da869e52af509cf237f35540ca/bin/busybox 
	linux-vdso.so.1 (0x00007ffc3f7ba000)
	libcrypt.so.1 => /lib64/libcrypt.so.1 (0x00007f63cef6a000)
	libm.so.6 => /lib64/libm.so.6 (0x00007f63cee87000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f63cec9a000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f63cf057000)
root@notubuntu:/home/krishjain/oci-sysext# ls -al /root/.local/share/oci-sysext/sysexts-rootfs/8dd682da869e52af509cf237f35540ca/usr/bin
total 476
drwxr-xr-x. 1 root root 1222 Dec 31  1969  .
drwxr-xr-x. 1 root root   56 Dec 31  1969  ..
lrwxrwxrwx. 1 root root   12 Dec 31  1969 '[' -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969 '[[' -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  awk -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  basename -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  bc -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  beep -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  bunzip2 -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  bzcat -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  bzip2 -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  cal -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  chrt -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  cksum -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  clear -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  cmp -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  comm -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  cryptpw -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  cut -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  dc -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  diff -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  dirname -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  dos2unix -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  du -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  env -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  expand -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  expr -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  factor -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  fallocate -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  find -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  flock -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  fold -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  free -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  fuser -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  groups -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  hd -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  head -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  hexdump -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  hostid -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  id -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  install -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  ipcrm -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  ipcs -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  killall -> /bin/busybox
lrwxrwxrwx. 1 root root   30 Dec 31  1969  ld.so -> ../../lib/ld-linux-x86-64.so.2
lrwxrwxrwx. 1 root root   12 Dec 31  1969  less -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  logger -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  lsof -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  lzcat -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  lzma -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  lzopcat -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  md5sum -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  microcom -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  mkfifo -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  mkpasswd -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  nl -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  nmeter -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  nohup -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  nproc -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  nsenter -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  od -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  passwd -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  paste -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  pgrep -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  pkill -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  pmap -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  printf -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  pstree -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  pwdx -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  readlink -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  realpath -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  renice -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  reset -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  resize -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  seq -> /bin/busybox
lrwxrwxrwx. 1 root root   12 Dec 31  1969  setsid -> /bin/busybox
.....

root@notubuntu:/home/krishjain/oci-sysext# /root/.local/share/oci-sysext/sysexts-rootfs/8dd682da869e52af509cf237f35540ca/bin/busybox
BusyBox v1.36.1 (2024-05-23 13:45:30 UTC) multi-call binary.
BusyBox is copyrighted by many authors between 1998-2015.
Licensed under GPLv2. See source distribution for detailed
copyright notices.

Usage: busybox [function [arguments]...]
   or: busybox --list[-full]
   or: busybox --install [-s] [DIR]
   or: function [arguments]...

	BusyBox is a multi-call binary that combines many common Unix
	utilities into a single executable.  Most people will create a
	link to busybox for each function they wish to use and BusyBox
	will act like whatever it was invoked as.

Currently defined functions:
	[, [[, add-shell, addgroup, adduser, adjtimex, arch, arping, ash, awk, base64, basename, bbconfig, bc, beep, bunzip2, bzcat, bzip2, cal, cat, chattr, chgrp, chmod, chown, chpasswd, chroot, chrt, cksum, clear, cmp, comm,
	cp, cpio, cryptpw, cut, date, dc, dd, delgroup, deluser, df, diff, dirname, dmesg, dnsdomainname, dos2unix, du, echo, ed, egrep, env, expand, expr, factor, fallocate, false, fgrep, find, findfs, flock, fold, free,
	fsync, fuser, getopt, getty, grep, groups, gunzip, gzip, hd, head, hexdump, hostid, hostname, id, inotifyd, install, ionice, iostat, ipcrm, ipcs, kill, killall, killall5, less, link, linux32, linux64, ln, logger, login,
	ls, lsattr, lsof, lzcat, lzma, lzop, lzopcat, md5sum, microcom, mkdir, mkfifo, mknod, mkpasswd, mktemp, more, mountpoint, mpstat, mv, netstat, nice, nl, nmeter, nohup, nologin, nproc, nsenter, od, passwd, paste, pgrep,
	pidof, ping, ping6, pipe_progress, pivot_root, pkill, pmap, printenv, printf, ps, pstree, pwd, pwdx, rdev, readahead, readlink, realpath, remove-shell, renice, reset, resize, rev, rm, rmdir, run-parts, sed, seq,
	setpriv, setserial, setsid, sh, sha1sum, sha256sum, sha3sum, sha512sum, shred, shuf, sleep, sort, split, stat, strings, stty, su, sum, sync, sysctl, tac, tail, tar, tee, test, time, timeout, top, touch, tr, traceroute,
	traceroute6, tree, true, truncate, tsort, tty, ttysize, tunctl, uname, unexpand, uniq, unix2dos, unlink, unlzma, unlzop, unxz, unzip, uptime, usleep, uudecode, uuencode, vconfig, vi, vlock, watch, wc, which, who,
	whoami, xargs, xxd, xzcat, yes, zcat

```

**Comment by krishjainx** [comment]
So for the unpatched oci-sysext build (just with PR #1 applied), behaving this way is understandable.

Anyway @89luca89 this PR can be merged! Working on another PR for getting the patching stuff (not unlike uhaul) integrated into oci-sysext :)

**Comment by krishjainx** [comment]
**TLDR:**

For the **current state of oci-sysext** (without the patching functionality I **am** integrating into oci-sysext from uhaul), this PR has everything: https://github.com/89luca89/oci-sysext/pull/1. I lost a good part of my evening because I forgot that we (Chainguard) build Wolfi using glibc, not musl . Basically stuff that doesn't use musl **mostly works** (but that's because it uses the hosts libraries). I say mostly works because stuff like apt works but bash doesn't want to run because " /lib64/libtinfo.so.6: no version information available" as it's trying to use my host library ie. **symbol conflict**.

The only thing that PR #1 does is make the image-source to diff against optional, so you can convert a whole OCI image to a systemd-sysext. 

I will make a PR for the patching stuff too (to use the "correct" matching libraries etc)

**Comment by 89luca89** [comment]
Thanks @krishjainx for this awesome work, you rock! :heart:
